src/Common/SipHash.h

```diff
@@ -229,6 +299,12 @@ class SipHash
         return res;
     }
 
+    std::pair<UInt64, UInt64> get128LoHi()
```

Comment:
In general, I think I've simplified the interface of SipHash quite a bit. However, I also added some helper functions in an effort to avoid repetition.

I think I've addressed every concern of yours, but let me go over them:
- `void get128Impl(char * out)`: I added comments to address the impact of endianness and made it private
- `void get128(T & lo, T & hi)`: I built a `getSipHash128AsLoHi` helper function on top of it, but as a free function
- I eliminated `is_array_class` and employed a different approach
- I moved `get64` before the `get128` methods